src/core/MIGRATION_EXAMPLES.md

```diff
@@ -0,0 +1,255 @@
+# Migration Examples
+
+This document is a list of examples of how to migrate plugin code from legacy
+APIs to their New Platform equivalents.
+
+## Configuration
+
+### Declaring config schema
+
+Declaring the schema of your configuration fields is similar to the Legacy Platform but uses the `@kbn/config-schema` package instead of Joi. This package has full TypeScript support, but may be missing some features you need. Let the Platform team know by opening an issue and we'll add what you're missing.
+
+```ts
+// Legacy config schema
+import Joi from 'joi';
+
+new kibana.Plugin({
+  config() {
+    return Joi.object({
+      enabled: Joi.boolean().default(true),
+      defaultAppId: Joi.string().default('home'),
+      index: Joi.string().default('.kibana'),
+      disableWelcomeScreen: Joi.boolean().default(false),
+      autocompleteTerminateAfter: Joi.number().integer().min(1).default(100000),
+    })
+  }
+});
+
+// New Platform equivalent
+import { schema } from '@kbn/config-schema';
+
+// This object should be exported from your server-side plugin's index.ts file.
+export const config = {
+  schema: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+    defaultAppId: schema.string({ defaultValue: true }),
+    index: schema.string({ defaultValue: '.kibana' }),
+    disableWelcomeScreen: schema.boolean({ defaultValue: false }),
+    autocompleteTerminateAfter: schema.number({ min: 1, defaultValue: 100000 }),
```

Comment:
**optional**: sometimes `schema.duration` from `@kbn/config-schema` is more appropriate

Reply:
It doesn't currently support `min`